kubespawner/spawner.py

```diff
@@ -2988,13 +2989,27 @@ async def _load_profile(self, slug, user_options):
             # are in the form data posted. This prevents users who may be authorized
             # to only use one profile from being able to access options set for other
             # profiles
-            for option_name, option in profile.get('profile_options').items():
-                chosen_option = user_options.get(option_name)
-                if not chosen_option:
+            for (user_selected_option_name,) in selected_profile_user_options.keys():
+                if (
+                    user_selected_option_name
+                    not in profile.get('profile_options').keys()
+                ):
                     raise ValueError(
-                        f'Expected option {k} for profile {slug}, not found in posted form'
+                        f'Expected option {user_selected_option_name} for profile {slug}, not found in posted form'
                     )
 
+            # Get selected options or default to a random option if none is passed
```

Comment:
yes! Good catch. A change of thought happened in the meantime and forgot to update this

```suggestion
            # Get selected options or default to the first option if none is passed
```